stan/math/prim/arr/err/is_matching_size.hpp

```diff
@@ -0,0 +1,29 @@
+#ifndef STAN_MATH_PRIM_ARR_ERR_IS_MATCHING_SIZE_HPP
+#define STAN_MATH_PRIM_ARR_ERR_IS_MATCHING_SIZE_HPP
+
+#include <stan/math/prim/scal/err/is_size_match.hpp>
+
+namespace stan {
+namespace math {
+
+/**
+ * Return <code>true</code> if two structures are the same size.
```

Comment:
[optional]
This could be clarified to indicate that it's just the top-level size for `std::vector`, but total size for `Eigen::Matrix`.   So a 2 x 3 array returns 2, but a 2 x 3 matrix will return 6.